core/trino-main/src/test/java/io/trino/sql/planner/TestRemoveRedundantJoinPlan.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.sql.planner.assertions.BasePlanTest;
+import io.trino.sql.planner.plan.JoinNode;
+import io.trino.sql.tree.GenericLiteral;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.trino.sql.planner.assertions.PlanMatchPattern.aggregation;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.filter;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.join;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.output;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.project;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.values;
+import static io.trino.sql.planner.plan.JoinNode.DistributionType.REPLICATED;
+import static io.trino.sql.planner.plan.JoinNode.Type.INNER;
+
+public class TestRemoveRedundantJoinPlan
+        extends BasePlanTest
+{
+    @Test
+    public void testSemiJoinWithEmptyRight()
+    {
+        assertPlan("WITH t(msg) AS (VALUES ROW(CAST(ROW(1, 2.0) AS ROW(x BIGINT, y DOUBLE))), ROW(CAST(ROW(3, 4.0) AS ROW(x BIGINT, y DOUBLE)))) " +
+                        "SELECT a.msg.x FROM t a WHERE a.msg.x in (SELECT 1 WHERE false)",
```

Comment:
nit: uppercase sql keywords:
```suggestion
                        "SELECT a.msg.x FROM t a WHERE a.msg.x IN (SELECT 1 WHERE false)",
```

Reply:
fixed